ironfist/src/cpp/registry_prefs.h

```diff
@@ -0,0 +1,42 @@
+#ifndef REGISTRY_PREFS_H
+#define REGISTRY_PREFS_H
+
+#include <Windows.h>
+#include <string>
+
+template<typename T> T read_registry_pref(const std::string& pref_key);
+
+template<> DWORD read_registry_pref(const std::string& pref_key)
+	{
+	DWORD Type;
+	DWORD cbData = 4;
+	DWORD Data = (DWORD)-1;
+	LPCSTR SubKey = "SOFTWARE\\New World Computing\\Heroes of Might and Magic 2\\1.0";
+	HKEY hKey = nullptr;
+	if(RegCreateKeyA(HKEY_LOCAL_MACHINE, SubKey, &hKey) == ERROR_SUCCESS)
+		{
+		RegQueryValueExA(hKey, pref_key.c_str(), 0, &Type, (LPBYTE)&Data, &cbData);
+		}
+
+	return Data;
+	}
+
+template<typename T> bool write_registry_pref(const std::string& pref_key, const T& value);
+
+template<> bool write_registry_pref(const std::string& pref_key, const DWORD& value)
+	{
+	DWORD cbData = 4;
+	LPCSTR SubKey = "SOFTWARE\\New World Computing\\Heroes of Might and Magic 2\\1.0";
+	HKEY hKey = nullptr;
+
+	if(RegOpenKeyExA(HKEY_LOCAL_MACHINE, SubKey, 0, 0xF003Fu, &hKey) != ERROR_SUCCESS)

```

Comment:
"0xF003Fu" is what we call a "magic number." It's not at all clear where it came from. Presumably this is a bitwise-or of several constants or a special value defined somewhere. I think it's easier to read when written as such.